type.go

```diff
@@ -182,16 +194,28 @@ func parseRequestType(t reflect.Type) (*requestType, error) {
 			return nil, errgo.Mask(err)
 		}
 
-		if f.Anonymous {
-			if tag.source != sourceBody && tag.source != sourceNone {
-				return nil, errgo.New("httprequest tag not yet supported on anonymous fields")
-			}
+		if f.Anonymous && tag.source != sourceNone {
+			taggedFieldIndex = f.Index
 		}
 		pt.fields = append(pt.fields, field)
 	}
 	return &pt, nil
 }
 
+// withinIndex reports whether the field index i0 should be considered
+// to be within" the field i1 - that is, whether i0 is prefixed by i1.
```

Comment:
Also this comment is confusing, if i0 is prefixed by i1 then I'd hav thought i1 was within i0 not vice versa. Can you clarify this please?